rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy

```diff
@@ -1049,6 +1059,68 @@ class ScheduledExecutionService implements ApplicationContextAware, Initializing
         claimed
     }
 
+    def reclaimAndScheduleJobByJob() {
```

Comment:
by default since the class is marked `@Transactional`, calling this method will create a new outer transaction.  Do you want to mark it `@NotTransactional`?

Reply:
Marking this method as `@NotTransactional` causes this function to explode with `org.springframework.dao.DataAccessResourceFailureException: Could not obtain current Hibernate Session; nested exception is org.hibernate.HibernateException: No Session found for current thread` when we attempt to `.refresh()` the `ScheduledExecution` between the first and second nested transactions.

I would kind of prefer to leave this as-is since we have done a lot of testing here. Do you see any issue with that?

Reply:
Just a couple of final notes here: the previous implementation worked as expected since `withNewTransaction` _always_ creates a separate, totally independent transaction using `PROPAGATION_REQUIRES_NEW`. However using `@NotTransactional` as you suggested should use slightly fewer resources so I went ahead and made that change.  

(See https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/tx-propagation.html#tx-propagation-requires_new for more detail.)